cinema/views.py

```diff
@@ -118,12 +156,25 @@
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get_permissions(self):
+        if self.action == "list" and self.request.user.is_authenticated:
+            self.permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
The `get_permissions` method in `OrderViewSet` sets `IsAdminOrIfAuthenticatedReadOnly` for the `list` action when the user is authenticated. Ensure this aligns with your application's permission requirements, as it might override the default `IsAuthenticated` permission.